### PR TITLE
Feature/adding post model and authentication

### PR DESCRIPTION
**Pull request by slevvvvin**

### CODE REVIEW

File: myDjangoBlog/urls.py [hunk 1]
```diff
@@ -1,21 +1,15 @@
...
 from django.contrib import admin
 from django.urls import path
+from django.urls import include
+
+
 
 urlpatterns = [
-    path('admin/', admin.site.urls),
-]
+
```
Comment (TinaArts) [inline myDjangoBlog/urls.py]: Please remove extra empty lines add new on in the end of code

**Review by TinaArts — CHANGES_REQUESTED** [review]
1. I think you in this case you don't need to separate applications - 'blog' and 'member'
2. Please install plugin for checking code by pep8 (like flake8) and fix all issues

File: blog/templates/add_category.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{% extends 'base.html'   %}
+
+{% block title %}
+Create A New Category
+{% endblock %}
+
+{% block content %}
+{% if user.is_authenticated %}
+    <h1>Add Category</h1>
+    <br/><br/>
```
Comment (TinaArts) [inline blog/templates/add_category.html]: For you next work with html - doesn’t use 'br' all time you can use css for styling blocks

File: blog/templates/category_list.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Blog Categories
+{% endblock %}
+
+{% block content %}
+        <h1>Blog Categories</h1>
+        {% for item in cat_menu_list %}
+            <h4><a class="dropdown-item" href="{% url 'category' item|slugify%}">
+                {{ item }}</a></h4>
+        {% endfor %}
+{% endblock %}
```
Comment (TinaArts) [inline blog/templates/category_list.html]: Please fix indents

File: blog/templates/home.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        - <a href ="{% url 'category' post.category|slugify%}">{{ post.category }}</a>
+        - {{ post.author.first_name}}
+        {{ post.author.last_name }} -
+        {{ post.post_date }}
+        {% if user.is_authenticated %}
+            - <a href="{% url 'update_post' post.pk %}">(Edit)</a>
+            <a href="{% url 'delete_post' post.pk %}">(Delete)</a>
+        {% endif %}
+        <br/>
+        {{ post.body|slice:":200"|safe }}</li>
```
Comment (TinaArts) [inline blog/templates/home.html]: You can use css instead of 'slice'  for cat text it should be faster

File: blog/forms.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+from django import forms
+from .models import Post, Category
+
+choices = Category.objects.all().values_list('name', 'name')
+choice_list = []
+
+for item in choices:
+    choice_list.append(item)
+
```
Comment (TinaArts) [inline blog/forms.py:3]: Move this logic to the model

File: blog/models.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django.urls import reverse
+
+
+class Category(models.Model):
+    name = models.CharField(max_length=250)
+
```
Comment (TinaArts) [inline blog/models.py:9]: Add fields that was in ER diagram

File: blog/models.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    def get_absolute_url(self):
+        return reverse('home')
+
+
+class Post(models.Model):
+    title = models.CharField(max_length=250)
+    title_tag = models.CharField(max_length=250)
+    author = models.ForeignKey(User, on_delete=models.CASCADE)
+    body = models.TextField()
+    post_date = models.DateField(auto_now_add=True)
```
Comment (TinaArts) [inline blog/models.py]: Use DataTime field

File: blog/models.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    category = models.CharField(max_length=250, default='uncategorized')
```
Comment (TinaArts) [inline blog/models.py]: Use field type by relationship between tbls

File: blog/templates/categories.html [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        {% if user.is_authenticated %}
+            - <a href="{% url 'update_post' post.pk %}">(Edit)</a>
+            <a href="{% url 'delete_post' post.pk %}">(Delete)</a>
+        {% endif %}
+        <br/>
+        {{ post.body|slice:":200"|safe }}</li>
+{% endfor %}
+</ul>
+{% else%}
+    <h2>Sorry, this page does not exist...</h2>
```
Comment (TinaArts) [inline blog/templates/categories.html]: Investigate how working with errors tmpls

File: blog/views.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+
+def CategoryListView(request):
+    cat_menu_list = Category.objects.all()
+    return render(request, 'category_list.html', {
+       'cat_menu_list': cat_menu_list
+        })
+
+
+def CategoryView(request, cats):
```
Comment (TinaArts) [inline blog/views.py]: Move logic with filtration by category to separate PR and use Based Views

File: blog/views.py [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+    })
+
+
+class UserRegisterView(generic.CreateView):
+    form_class = UserCreationForm
+    template_name = 'registration/register.html'
+    success_url = reverse_lazy('login')
+
+
+class ArticleDetailView(DetailView):
```
Comment (TinaArts) [inline blog/views.py]: Use 'Post' name in all cases

File: blog/views.py [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    model = Post
+    template_name = 'article_details.html'
+
+
+class AddPostView(CreateView):
```
Comment (TinaArts) [inline blog/views.py:54]: Read how you can use one Classe Based View for one entity

File: myDjangoBlog/urls.py [hunk 13]
```diff
@@ -1,21 +1,9 @@
...
 from django.contrib import admin
 from django.urls import path
+from django.urls import include
 
 urlpatterns = [
-    path('admin/', admin.site.urls),
+   path('admin/', admin.site.urls),
+   path('', include('blog.urls')),
+   path('blog', include('django.contrib.auth.urls')),
 ]
```
Comment (TinaArts) [inline myDjangoBlog/urls.py:9]: Change urls path

**Review by TinaArts — CHANGES_REQUESTED** [review]

File: blog/views.py (hunk 12 above)
Comment (slevvvvin) [reply blog/views.py:54]: will do in the next PR

File: blog/urls.py [hunk 14]
```diff
@@ -0,0 +1,14 @@
+from django.urls import path
+from .views import HomeView, PostDetailView, AddPostView, UpdatePostView,\
+    DeletePostView, AddCategoryView, CategoryView, UserRegisterView
+
+urlpatterns = [
+    path('category/<int:pk>/', CategoryView.as_view(), name='category'),
+    path('', HomeView.as_view(), name='home'),
```
Comment (TinaArts) [inline blog/urls.py:6]: This url should be first

File: blog/views.py [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+
+
+class PostDetailView(DetailView):
+    model = Post
+    template_name = 'post_details.html'
+
+    def get_context_data(self, *args, **kwargs):
+        cat_menu = Category.objects.all()
+        context = super(PostDetailView, self).get_context_data()
+        context["cat_menu"] = cat_menu
```
Comment (TinaArts) [inline blog/views.py]: What is cat_menu? 
Please use more obvious names for variables, like 'category_list'. 
Also you can rewrite lines 51 and 53 to one line and make code more clean

File: docker-compose.yml [hunk 16]
```diff
@@ -1,35 +1,27 @@
...
- data_volume:
+# data_volume:
 
 services:
   db:
     build: ./docker/mysql/
-    volumes:
-      - ./data_volume:/var/lib/mysql
+#    volumes:
+#      - ./data_volume:/var/lib/mysql
```
Comment (TinaArts) [inline docker-compose.yml]: Please remove all unused code

**Review by TinaArts — CHANGES_REQUESTED** [review]

File: myDjangoBlog/urls.py (hunk 1 above)
Comment (slevvvvin) [reply myDjangoBlog/urls.py]: done

File: blog/templates/add_category.html (hunk 2 above)
Comment (slevvvvin) [reply blog/templates/add_category.html]: Ok, I will

File: blog/templates/category_list.html (hunk 3 above)
Comment (slevvvvin) [reply blog/templates/category_list.html]: done

File: blog/templates/home.html (hunk 4 above)
Comment (slevvvvin) [reply blog/templates/home.html]: done

File: blog/forms.py (hunk 5 above)
Comment (slevvvvin) [reply blog/forms.py:3]: done

File: blog/models.py (hunk 6 above)
Comment (slevvvvin) [reply blog/models.py:9]: added Author field like in ERD

File: blog/templates/categories.html (hunk 9 above)
Comment (slevvvvin) [reply blog/templates/categories.html]: ok, I'll do it

File: blog/views.py (hunk 10 above)
Comment (slevvvvin) [reply blog/views.py]: done

File: blog/views.py (hunk 11 above)
Comment (slevvvvin) [reply blog/views.py]: done

File: blog/views.py (hunk 12 above)
Comment (slevvvvin) [reply blog/views.py:54]: and done

File: myDjangoBlog/urls.py (hunk 13 above)
Comment (slevvvvin) [reply myDjangoBlog/urls.py:9]: done

File: blog/urls.py (hunk 14 above)
Comment (slevvvvin) [reply blog/urls.py:6]: done

File: blog/views.py (hunk 15 above)
Comment (slevvvvin) [reply blog/views.py]: done

File: docker-compose.yml (hunk 16 above)
Comment (slevvvvin) [reply docker-compose.yml]: done

File: blog/views.py [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+
+
+class UpdatePostView(UpdateView):
+    model = Post
+    form_class = EditForm
+    template_name = 'update_post.html'
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(UpdatePostView, self).get_context_data()
+        context["category_list"] = Category.objects.all()
```
Comment (TinaArts) [inline blog/views.py:83]: will move to the tag on next PR

File: docker-compose.prod.yml [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+ # static_wd volume will be mounted to both nginx and django gunicorn services.
+ static_volume:
+ # mysql data volume will be mounted to mysql services
+ data_volume:
+
+services:
+  db:
+    build: ./docker/mysql/
+    volumes:
+      - ./data_volume:/var/lib/mysql
```
Comment (TinaArts) [inline docker-compose.prod.yml]: Remove this volumes, you don't need it

**Review by TinaArts — APPROVED** [review]

**Pull request merged**